### ISSUE TITLE
Need help importing OpenCV

### ISSUE THREAD

**Issue by ShikyC** [BroGamer4256/TaikoArcadeLoader#6]
Hi, I need some help to import OpenCV as a subproject to this project. What I've done is:

1. Adding a `opencv.wrap` file in the `subprojects` folder:
  ```
[wrap-file]
directory = opencv-2.4.13.7
source_url = https://github.com/opencv/opencv/archive/refs/tags/2.4.13.7.zip
source_filename = opencv-2.4.13.7.zip
source_hash = F1CD0E7760D4D69CA78A84EA42468E0427F5AF56E256D6D5BCBE3AD76A6CB1B2

[provide]
opencv = opencv_dep
  ```

2. Modify the `meson.build` file to add the following content:

```
cmake = import('cmake')
opt_var = cmake.subproject_options()
opt_var.set_override_option('WITH_CUDA', '0')
opt_var.set_override_option('WITH_CUFFT', '0')
opt_var.set_override_option('WITH_CUBLAS', '0')
opt_var.set_override_option('ENABLE_DYNAMIC_CUDA', '0')
opt_var.set_override_option('BUILD_opencv_gpu', '0')
opencv_subproj = cmake.subproject('opencv', options: opt_var)
opencv_dep = opencv_subproj.dependency('opencv')
```

However when building with `make setup dist-no-7z`, I always get this error, despite that I have turned off the CUDA flags in the settings.

```subprojects/opencv-2.4.13.7/meson.build:1:0: ERROR: 'cuda' compiler binary not defined in cross or native file```

When I check the generated `meson.build` file for OpenCV in the `build` folder, I found that the first line contains a strange `cuda` compiter option, which should not exist:

```
project('OpenCV', 'c', 'cpp', 'cuda')
```

Can someone please help me on this?

**Comment by vixen256** [comment]
You need to add a cuda compiler to `cross-mingw-64.txt` by adding a line in the `binaries` section with the content `cuda = 'nvcc'`

**Comment by ShikyC** [comment]
Thanks! I'll try it out.

**Comment by ShikyC** [comment]
Hi @BroGamer4256 sorry to bother again but I found the CUDA issues was still hard to resolve even with nvcc installed and configured, so I switch to a newer OpenCV version that does not require CUDA. But this time it's the OpenCL glitching. 

Similar configuration, disabled OpenCL flag, but got `subprojects/opencv-3.4.20/meson.build:167:0: ERROR: File /home/TaikoArcadeLoader/build/subprojects/opencv-3.4.20/__CMake_build/modules/core/opencl_kernels_core.hpp does not exist.`

It looks like no matter how I set `WITH_OPENCL` flag values, meson always generates the `meson.build` file with all the OpenCL dependencies. However, it doesn't produce the corresponding files to the `build` folder. I found a similar issue [here](https://github.com/mesonbuild/meson/issues/12554) but I don't know if it's a meson or OpenCV issue (this problem seem to persist from OpenCV 3.4.20 to the latest 4.9)

I really want to implement the QR code reading function for TAL. Instead of buiding from scratch, is there an easier way to use submodules?
